File: week08/11/1/P1.c
#include <stdio.h>
#include <stdlib.h>

int solution(int K, int A[], int N){
    int result = 0;
    int cur = 0;
    for (int i = 0; i < N; i++){
        if (cur + A[i] >= K){
            result++;
            cur = 0;
        }
        else{
            cur = cur + A[i];
        }
    }
    return result;
}

int main(void){
    int K = 4;
    int A[] = {1, 2, 3, 4, 1, 1, 3};
    int N = 7;
    printf("%d\n", solution(K,A,N));
    return 0;
}

File: week13/19/P1.py
import heapq

class Solution(object):
    def lastStoneWeight(self, stones):
        maxHeap = []
        for i in stones:
            heapq.heappush(maxHeap, -i)
        while (len(maxHeap) > 1):
            first = heapq.heappop(maxHeap)
            second = heapq.heappop(maxHeap)
            if (first < second):
                heapq.heappush(maxHeap, first - second)
        if (len(maxHeap) > 0):
            return -maxHeap[0]
        return 0

if __name__ == "__main__":
    result = Solution()
    stones = [2,7,4,1,8,1]
    print(result.lastStoneWeight(stones))
    stones = [1]
    print(result.lastStoneWeight(stones))